File: backend-php/api/model-Eduardo/Animal.php
<?php

/**
 * @author Eduardo
 */
class Animal {
   
    private $raca;
    private $nome;
    private $idade;
   
    public function get_raca() {
        return $this->raca;
    }

    public function get_nome() {
        return $this->nome;
    }

    public function get_idade() {
        return $this->idade;
    }

    public function set_raca($raca) {
        $this->raca = $raca;
        return $this;
    }

    public function set_nome($nome) {
        $this->nome = $nome;
        return $this;
    }

    public function set_idade($idade) {
        $this->idade = $idade;
        return $this;
    }

}

File: AdotePetJavaProjetct/src/adotepet/Animal.java
package adotepet;

import adotepet.fabrica.ITipoAnimal;

public class Animal implements ITipoAnimal{
    private String raca, nome, tipo;
    private int idade;
    
    public Animal(String tipo, String raca, String nome, int idade){
        setTipo(tipo);
        setRaca(raca);
        setNome(nome);
        setIdade(idade);
    }

    public String getNome(){
        return nome;
    }

    public void setNome(String nome){
        this.nome=nome;
    }

    public String getRaca(){
        return raca;
    }

    public void setRaca(String raca){
        this.raca=raca;
    }

    public int getIdade(){
        return idade;
    }

    public void setIdade(int idade){
        this.idade=idade;
    }

    public String getTipo(){
        return tipo;
    }

    public void setTipo(String tipo){
        this.tipo=tipo;
    }
    
    public void criaAnimal(){
        System.out.println("Animal criado");
    }
}